generate/generate.go

```diff
@@ -152,8 +152,17 @@ func (g *Generator) generateProtoc(req plugin.CodeGeneratorRequest, gen config.G
 	// a formatted list of what is in req.FileToGenerate.
 	protoFilenames := []string{}
 
+	// Make copies of the req.ProtoFile so we don't accidentally change
+	// the values of the pointers which will affect all other protoc
+	// generator runs.
+	protoFiles := make([]*desc.FileDescriptorProto, len(req.ProtoFile))
```

Comment:
I am sure there is a better way to do this..?

Reply:
If you only wanted to copy the array, you could do `list2 := append([]T{}, list1...)`. However, you want to copy each pointer's backing data too, so I don't think there's a short-hand for this.

Perhaps we could make the slice a `[]desc.FileDescriptorProto` instead. But I don't think that's worth it, just to avoid three lines of code.